src/result_processor.h

```diff
@@ -311,6 +314,26 @@ void PipelineAddCrash(struct AREQ *r);
   *******************************************************************************************************************/
  ResultProcessor *RPMaxScoreNormalizer_New(const RLookupKey *rlk);
 
+ /*******************************************************************************************************************
+  *  Hybrid Merger Result Processor
+  *
+  * Merges results from two upstream processors using a hybrid scoring function.
+  * Takes results from both upstreams and applies the provided function to combine their scores.
+  *******************************************************************************************************************/
+
+ // Parameters:
+ //   scores: Array of score values from each upstream
+ //   hasScores: Array of boolean flags indicating if each score is available
+ //   numUpstreams: Number of upstream processors
+ // Returns: Combined score calculated from the array of optional input scores
+ typedef double (*HybridScoringFunction)(double *scores, bool *hasScores, size_t numUpstreams);
+
+
+ ResultProcessor *RPHybridMerger_New(HybridScoringFunction hybridScoring_function,
```

Comment:
Intuitively I would implement the RRF-hybrid-scoring-function in this PR as well so this RP will be usable after this PR.
I assume you didn't do that in order not to blow up the PR.. Legit consideration but it will yield some tail for the PR